CMakeLists.txt

```
@@ -12,16 +12,22 @@ set(GCF_IGNORE_LIST "" CACHE STRING
     "Semi colon separated list of directories to ignore")
 set(GCF_CLANGFORMAT_MINIMAL_VERSION "0.0.0" CACHE STRING
     "The minimal required version of clang-format")
-set(GCF_FORCE_OVERWRITE OFF CACHE BOOL
-    "If true, always overwrite pre-commit hook and script")
+option(GCF_FORCE_OVERWRITE
+       "If true, always overwrite pre-commit hook and script"
+       OFF)
 set(GCF_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/git-cmake-format.py)
+if(NOT GCF_GIT_TOP_LEVEL_DIR)
+  message(FATAL_ERROR
+          "Need to specify the top-level directory of the git repository "
+          "into which the hooks should be installed.")
+endif()
```

```suggestion
set(GCF_GIT_TOP_LEVEL_DIR "${CMAKE_SOURCE_FOLDER}" CACHE STRING "The default top level is the CMAKE_SOURCE_FOLDER")
if("${GCF_GIT_TOP_LEVEL_DIR}" STREQUAL "${CMAKE_SOURCE_FOLDER}")
  message(WARNING
          "git-cmake-format uses the current CMAKE_SOURCE_FOLDER, which may affect more than expectation")
endif()
```
For me, warning is enough for it?